tests/PHPStan/Type/MixedTypeTest.php

```diff
@@ -164,4 +166,73 @@ public function testIsSuperTypeOf(MixedType $type, Type $otherType, TrinaryLogic
 		);
 	}
 
+	public function dataSubstractedIsArray(): array
+	{
+		return [
+			[
+				new MixedType(),
+				new ArrayType(new IntegerType(), new StringType()),
+				TrinaryLogic::createNo(),
```

Comment:
subtracting only `array<int, string>` could still leave e.g. a `array<string, mixed>` within the `mixed`.
therefore I am wondering, whether the only case where `isArray` can say NO is, when a general `array` is substracted?

Reply:
Exactly. Switching the callee and the argument works even when more than just an array are subtracted, like array|null for example.

Reply:
I just see https://github.com/phpstan/phpstan-src/pull/1656#discussion_r957025938, so I think we are on the same page.. will adjust the PR.